casper/src/test/scala/coop/rchain/casper/engine/InitializingSpec.scala

```diff
@@ -116,10 +116,8 @@ class InitializingSpec extends WordSpec with BeforeAndAfterEach {
         // Assert requested messages for the state and fork choice tip
         _        = assert(transportLayer.requests.size == expectedRequests.size)
         messages = transportLayer.requests.map(_.msg)
-        _ = messages.zip(expectedRequests).map {
-          case (msg1, msg2) => assert(msg1 == msg2)
-        }
-        _ = transportLayer.reset()
+        _        = assert(messages.forall(m => expectedRequests.exists(e => e == m)))
+        _        = transportLayer.reset()
```

Comment:
This works but it's not precise enough. We should expect all requests correctly and not just part of it.

Order can be non-deterministic so comparing sets can be enough.
`assert(messages.toSet == expectedRequests.toSet)`

Maybe also delay in line 102 of 1 second is not enough to create the same requests. :thinking:

Reply:
`forall` is not enough?

Reply:
Not sure what you mean.

Reply:
P.S. Overall, I'm not sure what this test is supposed to check. If just Initializing engine why all other stuff need to be created, like everything (block store, dag store, casper, ...) and also in the same way as in normal node operation, with LMDB and on file system.

Reply:
I changed delay to 5 seconds. This should be enough for now.
https://github.com/rchain/rchain/pull/3134/files#diff-91178b6525f827b57ac86c3d7c1a3cd1R102